### PR TITLE
Tweaks to existing 2fa

### PR DESCRIPTION
**Pull request by BeckaL**
This PR is a preliminary PR ahead of some work to allow exemptions for 2fa ([trello](https://trello.com/c/qijNj4uh/433-allow-admins-super-admins-gds-only-to-exempt-users-from-2sv)).

It:
* Changes terminology of 'flagging 2sv' to 'mandating 2sv' in code to make the code clearer 
* Introduces a small styling change (changes a bullet point to a paragraph) - this paves the way for planned additions to this section for exempting 2sv, where it makes more sense to have paragraphs rather than bullet points.
* Adds an explicit message for when a user has two step verification flagged, but not yet set up. It also removes the checkbox to toggle two-step verification in this case, because exemptions will be the way to achieve this, and it's somewhat a loophole that you can remove 2sv when a user has had 2sv mandated but they haven't yet set it up, but not when they have set it up.

Screenshots to demonstrate points 2 and 3 above (note difference in row 2 for point 2, and row 3 for point 3), taken from the display of 'account security' section on the edit user page:

| | Current display | Changes in this PR |
|---|---|---|
| When a user has not had 2sv mandated yet|Screenshot 2023-01-31 at 10 04 51|Screenshot 2023-01-31 at 09 59 28|
| When a user has had 2sv mandated but has not yet set it up |Screenshot 2023-01-31 at 10 05 47|Screenshot 2023-01-31 at 09 58 34| 
| When a user has set up 2sv|Screenshot 2023-01-31 at 10 08 20|Screenshot 2023-01-31 at 09 58 48|

⚠️ This repo is Continuously Deployed: make sure you [follow the guidance](https://docs.publishing.service.gov.uk/manual/development-pipeline.html#merge-your-own-pull-request) ⚠️

Follow [these steps](https://guides.rubyonrails.org/upgrading_ruby_on_rails.html) if you are doing a Rails upgrade.

### CODE REVIEW

File: app/views/users/_form_fields.html.erb [hunk 1]
```diff
@@ -44,17 +44,17 @@
...
             <%= link_to 'Reset 2-step verification',
               reset_two_step_verification_user_path(@user),
               data: { confirm: 'Are you sure?' },
               method: :patch
               %><br/>
             Allows user to sign in without a verification code.<br/>
             User will be prompted to set up 2-step verification again the next time they sign in.
-          </li>
-        </ul>
+          </p>
```
Comment (brucebolt) [inline app/views/users/_form_fields.html.erb:55]: This `<p>` block doesn't need to be indented here (it should be the same as the indentation level on line 46).

File: app/views/users/_form_fields.html.erb [hunk 2]
```diff
@@ -44,17 +44,17 @@
...
+      <% elsif f.object.require_2sv? %>
+          <p>2-step verification required but not set up</p>
```
Comment (brucebolt) [inline app/views/users/_form_fields.html.erb:57]: Same here, the indentation can be moved to the left by one level.

File: test/integration/mandating_two_step_verification_test.rb [hunk 3]
```diff
@@ -116,4 +118,26 @@ def assert_admin_can_remove_2sv_requirement_without_notifying_user(admin, user)
       assert page.has_no_text? "Ask user to set up 2-step verification"
     end
   end
+
+  context "when a user has already been flagged for 2sv" do
```
Comment (brucebolt) [inline test/integration/mandating_two_step_verification_test.rb:122]:
```suggestion
  context "when a user has already been mandated for 2sv" do
```

**Review by brucebolt — APPROVED** [review]
These changes look good to me.

I've just added a few comments about indentation and a use of the word `flagged` in a test.

File: test/integration/mandating_two_step_verification_test.rb (hunk 3 above)
Comment (BeckaL) [reply test/integration/mandating_two_step_verification_test.rb:122]: Good spot

**Pull request merged**